pex/bin/pex.py

```diff
@@ -807,7 +807,7 @@ def to_python_interpreter(full_path_or_basename):
                 )
             )
 
-    interpreter = min(interpreters) if interpreters else None
+    interpreter = PythonInterpreter.safe_min(interpreters) if interpreters else None
```

Comment:
It felt wrong to put this logic in `__lt__`. I want to call attention to the magic.

Because I deleted `__lt__`, calling `min()` will fail.